src/zcash/NoteEncryption.cpp

```diff
@@ -168,37 +160,34 @@ std::optional<SaplingEncPlaintext> AttemptSaplingEncDecryption(
     SaplingEncPlaintext plaintext;
 
     if (crypto_aead_chacha20poly1305_ietf_decrypt(
-        plaintext.begin(), NULL,
-        NULL,
-        ciphertext.begin(), SAPLING_ENCCIPHERTEXT_SIZE,
-        NULL,
-        0,
-        cipher_nonce, K) != 0)
-    {
+            plaintext.begin(),
+            NULL,
+            NULL,
+            ciphertext.begin(),
+            SAPLING_ENCCIPHERTEXT_SIZE,
+            NULL,
+            0,
+            cipher_nonce,
+            K)
+        != 0) {
         return std::nullopt;
     }
```

Comment:
This looks a bit odd. I think that with my suggested options it would end up as
```c++
    if (crypto_aead_chacha20poly1305_ietf_decrypt(
                plaintext.begin(),
                NULL,
                NULL,
                ciphertext.begin(),
                SAPLING_ENCCIPHERTEXT_SIZE,
                NULL,
                0,
                cipher_nonce,
                K
        ) != 0) {
        return std::nullopt;
    }